### PR TITLE
[RN] Make the calendar the default tab when there are calendar entries fetched

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: react/features/calendar-sync/middleware.js [hunk 1]
```diff
@@ -66,11 +66,9 @@ CALENDAR_ENABLED
                     && store.dispatch(addKnownDomains(knownDomains));
             }
 
-            const result = next(action);
-
             _fetchCalendarEntries(store, false, false);
 
-            return result;
+            return next(action);
```
Comment (saghul) [inline react/features/calendar-sync/middleware.js]: is this change necessary?

File: react/features/welcome/components/WelcomePageLists.js [hunk 2]
```diff
@@ -74,15 +80,35 @@ class WelcomePageLists extends Component<Props> {
...
+ * @param {Object} state - The redux state.
+ * @protected
+ * @returns {{
+ *     _hasCalendarEntries: boolean
+ * }}
+ */
+function _mapStateToProps(state: Object) {
+    const { events } = state['features/calendar-sync'];
+
+    console.log('ZB:_mapStateToProps', events);
```
Comment (saghul) [inline react/features/welcome/components/WelcomePageLists.js]: leftover

File: react/features/welcome/components/WelcomePageLists.js [hunk 3]
```diff
@@ -74,15 +80,35 @@ class WelcomePageLists extends Component<Props> {
...
+
+    return {
+        _hasCalendarEntries: events && events.length
```
Comment (saghul) [inline react/features/welcome/components/WelcomePageLists.js]: maybe wrap in a `Boolean(...)` just to make sure we always pass a boolean and not undefined some times

File: react/features/welcome/components/WelcomePageLists.js (hunk 2 above)
Comment (zbettenbuk) [reply react/features/welcome/components/WelcomePageLists.js]: Shit, I forgot to do a push after the final change :)

File: react/features/calendar-sync/middleware.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/calendar-sync/middleware.js]: I want the calendar entries to be available before the APP_WILL_MOUNT action bubbles all the way up so then there is only a slight delay between rendering the welcome screen and getting the entries

Comment (saghul) [reply react/features/calendar-sync/middleware.js]: 👍

**Comment by saghul** [comment]
Since the physical iOS device seems to be disconnected (cause for the iOS test failure) I tested it myself, *artisanally*. 

Good work @zbettenbuk !
